projects/plugins/jetpack/modules/search/instant-search/components/dom-event-handler.jsx

```diff
@@ -182,15 +183,52 @@ export default class DomEventHandler extends Component {
 		}
 	};
 
-	scrollOverlayToTop = event => {
+	fixBodyScroll = event => {
 		// NOTE: the propertyName need to be aligned with the animation
 		if ( event?.propertyName !== 'opacity' ) {
 			return;
 		}
-		// @see https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollTo
-		window?.scrollTo( 0, 0 );
+
+		if ( this.props.isVisible ) {
+			this.preventBodyScroll();
+			// This is to fix search input on the overlay is not visible on mobile devices.
+			// @see https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollTo
+			window?.scrollTo( 0, 0 );
+		} else {
+			this.restoreBodyScroll();
+		}
 	};
 
+	preventBodyScroll() {
+		this.top = parseInt( window.scrollY ) || 0;
+		// Keep body at the same position when overlay is open.
+		document.body.style.top = `-${ this.top - this.getWpAdminBarComputedHeight() }px`;
+		// Make body in the center.
+		document.body.style.left = 0;
+		document.body.style.right = 0;
+		// Make body not scrollable.
+		document.body.style.position = 'fixed';
+	}
+
+	restoreBodyScroll() {
+		// Restore body scroll.
+		document.body.style.top = '';
+		document.body.style.left = '';
+		document.body.style.right = '';
+		document.body.style.position = '';
+		// Restore body position.
+		window.scrollTo( 0, this.top );
+	}
+
+	getWpAdminBarComputedHeight() {
```

Comment:
Can we add some comments explaining why this wp-admin bar needs to be taken into account?

Would we face the same issue if the customer site had added their own fixed bar at the top of the page?

Reply:
>Can we add some comments explaining why this wp-admin bar needs to be taken into account?

This is a good question. The problem is wp admin bar offset html to be sticky, which is very rare nowadays. That's the reason why we consider this separately. Otherwise the page would not keep at the same position because of the offset. But I'm sure there's better way to get the offset.

>Would we face the same issue if the customer site had added their own fixed bar at the top of the page?

I did consider this problem. And looked at the P2 bar, which is fine, not like the wp admin bar which does offset to html...

Reply:
changed to use `document.documentElement.scrollHeight - document.body.scrollHeight`